UserService/Controllers/UserController.cs

```diff
@@ -71,4 +72,28 @@ public IActionResult GetUser([FromRoute]string id)
             return StatusCode(500, $"An unexpected error occurred: {ex.Message}");
         }
     }
+
+    [HttpPut("{id}")]
+    public IActionResult UpdateUser([FromRoute] string id, [FromBody]UserDtoUpdate userUpdateDto)
+    {
+        if(!ModelState.IsValid) return BadRequest();
+
+        try
+        {
+            if(!_userRepository.UserExists(id)) return NotFound();
+
+            var user = _mapper.Map<User>(userUpdateDto);
```

Comment:
what happens to the other fields that don't exist in the userupdatedto? are they set to null in the user model? does that mean we could overwrite data unintentionally?

Reply:
Native data types like int, float take 0 an 0,0 successively
Object like string, it takes null.
Maybe we should first track the last version of our object, so we can keep the values ​​of the fields unchanged